lib/models/package-info-cache/index.js

```diff
@@ -626,15 +628,21 @@ class PackageInfoCache {
       }
     }
 
-    // read addon modules from node_modules. We read the whole directory
-    // because it's assumed that npm/yarn may have placed addons in the
-    // directory from lower down in the project tree, and we want to get
-    // the data into the cache ASAP. It may not necessarily be a 'real' error
-    // if we find an issue, if nobody below is actually invoking the addon.
-    let nodeModules = this._readNodeModulesList(path.join(realPath, 'node_modules'));
+    if (pkgInfo.mayHaveAddons) {
+      // read addon modules from node_modules. We read the whole directory
+      // because it's assumed that npm/yarn may have placed addons in the
+      // directory from lower down in the project tree, and we want to get
+      // the data into the cache ASAP. It may not necessarily be a 'real' error
+      // if we find an issue, if nobody below is actually invoking the addon.
+      let nodeModules = this._readNodeModulesList(path.join(realPath, 'node_modules'));
 
-    if (nodeModules) {
-      pkgInfo.nodeModules = nodeModules;
+      if (nodeModules) {
+        pkgInfo.nodeModules = nodeModules;
+      }
+    } else {
+      // will not have addons, so even if their are node_modules here, we can
```

Comment:
'there are', not 'their are'

Reply:
english, how does it even work

Reply:
:-) My mom was an English major and writer, what can I say? It's inbred. ;-)